### ISSUE TITLE
Allow to compile on 64-bit

### ISSUE THREAD

**Issue by fxcoudert** [ascii-boxes/boxes#38]
The source code of boxes is actually 64-bit compatible, but it currently does not build in 64-bit on macOS. The reason: the `-ansi` flag is passed to the compiler, although non-ANSI (non-C89) functions are allowed (such as strings functions like `strdup`). Thus the `-ansi` flag should be removed.

Another unrelated issue is that `src/parser.y` uses `strcasecmp` without including the necessary `strings.h` system header.

The attached patch fixes both issues, and allows boxes to compile in 64-bit mode, which is the default on macOS (most computers nowadays are 64-bit, and many do not carry 32-bit compatibility anymore).

---

```
diff --git a/src/Makefile b/src/Makefile
index c2656df..db19d29 100644
--- a/src/Makefile
+++ b/src/Makefile
@@ -62,7 +62,7 @@ boxes.exe: $(ALL_OBJ)
 
 
 flags_unix:
-       $(eval CFLAGS := -ansi -I. -Iregexp -Wall -W $(CFLAGS_ADDTL))
+       $(eval CFLAGS := -I. -Iregexp -Wall -W $(CFLAGS_ADDTL))
        $(eval LDFLAGS := -Lregexp $(LDFLAGS_ADDTL))
        $(eval BOXES_EXECUTABLE_NAME := boxes)
        $(eval ALL_OBJ := $(GEN_SRC:.c=.o) $(ORIG_NORM:.c=.o))
diff --git a/src/parser.y b/src/parser.y
index c9acfbd..a32ef36 100644
--- a/src/parser.y
+++ b/src/parser.y
@@ -27,6 +27,7 @@
 #include <stdio.h>
 #include <stdlib.h>
 #include <string.h>
+#include <strings.h>
 #include "shape.h"
 #include "boxes.h"
 #include "tools.h"
```

**Comment by tsjensen** [comment]
Thank you for this patch!
If you'd like, you could create a pull request with 2 commits (the missing `#include` and the changed flag). Then you'd get contributor credits. If it's too much work, I can do it for you, of course.

*boxes* is generally a 32-bit application, which may still be compiled on 64-bit systems by adding some flags to the make command (from our [faq](http://boxes.thomasjensen.com/docs/faq.html#q5)):

    make CFLAGS_ADDTL=-m32 LDFLAGS_ADDTL=-m32

Did you try that? It's just that I'm curious to understand your use case. Since *boxes* is older than most 64-bit systems, just taking away the `-ansi` flag will probably not make it into a 64-bit application. So if the goal is just to make it compile, then it would be interesting to know why the method from the faq did not work for you. Thanks!

**Comment by fxcoudert** [comment]
Forcing 32-bit compilation with added flags works, but I wanted to build the app as a 64-bit binary. Modern systems are 64-bit by default, and over time the 32-bit environments are being removed or not installed by default. As far as I can tell, most of the source code of boxes will work fine compiled in 64-bit, if string indexes stay within 32-bit bounds (which is the most common case).

And no, removing `-ansi` does not make it automatically 64-bit compatible, but the `-ansi` flag is wrong nonetheless: there are plenty of non-C89 functions used in the code (including some BSD and POSIX functions).

PS: I would rather you commit the modifications yourself, though thanks for offering credit for the bug fixes.

**Comment by tsjensen** [comment]
Ok, gotcha. Thanks for taking the time to explain.
In order to check if it works on your 64-bit system, you can run the [test suite](http://boxes.thomasjensen.com/testing.html) via `make test`.

Do you need a new *boxes* release to get these changes into homebrew?

**Comment by fxcoudert** [comment]
With 64-bit compilation (default on my system after your patch), the testsuite runs fine except for:

```
Running test case: 004_missing_option_argument.txt
    Invoking: boxes -s
1c1
< boxes: option requires an argument -- s
---
> boxes: option requires an argument -- 's'
Error in test case: 004_missing_option_argument.txt (top: actual; bottom: expected)
```

and

```
Running test case: 065_size_missing_argument.txt
    Invoking: boxes -s
1c1
< boxes: option requires an argument -- s
---
> boxes: option requires an argument -- 's'
Error in test case: 065_size_missing_argument.txt (top: actual; bottom: expected)
```

**Comment by fxcoudert** [comment]
Yes, a new release would help. Can you please notify me here when you publish one? I would then update Homebrew.

**Comment by tsjensen** [comment]
I made an attempt at adjusting the two test cases that failed for you. Could you try if it works now? It would be nice if the whole test suite ran without errors.

I will do a new release then. This is a manual process, so it will take a few days. Expect it to be ready on the weekend. Thank you for your work supporting *boxes* on Homebrew!

**Comment by fxcoudert** [comment]
Now: `84 tests executed, 84 successful, 0 failed.`

(Although the testsuite does not run natively, because it requires GNU sed instead of BSD sed. But if one changes that, it now runs perfectly.)

**Comment by tsjensen** [comment]
@fxcoudert Released as [1.2](https://github.com/ascii-boxes/boxes/releases/tag/v1.2) ([release notes](http://boxes.thomasjensen.com/2017/03/boxes-v1.2-released))
